### PR TITLE
`settings/accounts` Ionio restoration

### LINKED ISSUE #440: User should be able to import Ionio accounts restoration file in `settings/accounts` page
related to #436 

https://github.com/vulpemventures/marina/pull/436#issuecomment-1470389783

### PR DESCRIPTION
**Pull request by louisinger**
This PR adds a new page in `settings/accounts` with a form letting to restore Ionio accounts.

the restore process:
1. creates the accounts if the wallet doesn't know them (that's why need the password input)
2. persists the script details associated with Ionio artifacts
3. triggers an update process on the restored scripts

it closes #440 

@tiero @bordalix please review

### CODE REVIEW

File: src/extension/onboarding/end-of-flow/index.tsx [hunk 1]
```diff
@@ -85,13 +85,12 @@ const EndOfFlowOnboarding: React.FC = () => {
       const restoration = await onboardingRepository.getRestorationJSONDictionary();
       if (restoration) {
         const accountsToRestore: Record<string, Array<NetworkString>> = {};
-        for (const [network, restorations] of Object.entries(restoration)) {
+        for (const restorations of Object.values(restoration)) {
           const accountNames = restorations.map((r) => r.accountName);
           for (const accountName of accountNames) {
             if (!accountsToRestore[accountName]) {
-              accountsToRestore[accountName] = [];
+              accountsToRestore[accountName] = ['liquid', 'regtest', 'regtest'];
```
Comment (bordalix) [inline src/extension/onboarding/end-of-flow/index.tsx:92]: I think you meant
```suggestion
              accountsToRestore[accountName] = ['liquid', 'testnet', 'regtest'];
```

File: src/extension/onboarding/wallet-restore/index.tsx [hunk 2]
```diff
@@ -7,7 +7,8 @@ import { MnemonicField } from './mnemonic-field';
 import OnboardingForm from '../onboarding-form';
 import { init } from '../../../domain/repository';
 import { validateMnemonic } from 'bip39';
-import type { RestorationJSON, RestorationJSONDictionary } from '../../../application/account';
+import type { RestorationJSONDictionary } from '../../../application/account';
+import { checkRestorationDictionary } from '../../../application/account';
```
Comment (bordalix) [inline src/extension/onboarding/wallet-restore/index.tsx:11]: I personally prefer this format:
```suggestion
import { checkRestorationDictionary, type RestorationJSONDictionary } from '../../../application/account';
```

Comment (louisinger) [reply src/extension/onboarding/wallet-restore/index.tsx:11]: let's keep import type for consistency with other files (or change all...)

**Comment by bordalix** [comment]
If I click Download without a previously created Ionio account, the button doesn't work and throws an exception.

Screenshot 2023-03-29 at 11 14 06

Screenshot 2023-03-29 at 11 14 39

**Comment by tiero** [comment]
How do I test the restoration if I can't dowload in the first place?

**Comment by louisinger** [comment]
> How do I test the restoration if I can't dowload in the first place?

was a WIP commit, my bad. Just pushed the final version with the `Toast` notification for error.

**Comment by tiero** [comment]
Restoring at onboarding I have Marina hanging

**Comment by tiero** [comment]
also restoring JSON from Settings > Accounts hangs

image

**Pull request merged**